src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/contactperson/ContactPersonSynchronisationService.kt

```diff
@@ -50,23 +52,27 @@ class ContactPersonSynchronisationService(
 
   suspend fun personRestrictionUpserted(event: PersonRestrictionEvent) {
     val telemetry =
-      mutableMapOf("nomisPersonId" to event.personId, "dpsContactId" to event.personId, "nomisPersonRestrictionId" to event.visitorRestrictionId)
+      mutableMapOf<String, Any>("nomisPersonId" to event.personId, "dpsContactId" to event.personId, "nomisPersonRestrictionId" to event.visitorRestrictionId)
+    val telemetrySuffix = "contactperson-person-restriction-synchronisation"
 
     if (event.doesOriginateInDps()) {
       telemetryClient.trackEvent(
-        "contactperson-person-restriction-synchronisation-created-skipped",
+        "$telemetrySuffix-upserted-skipped",
         telemetry,
       )
     } else {
-      mappingApiService.getByNomisPersonRestrictionIdOrNull(nomisPersonRestrictionId = event.visitorRestrictionId)?.also {
-        // TODO - this is an upsert so this should be treated as the update flow
-        telemetryClient.trackEvent(
-          "contactperson-person-restriction-synchronisation-created-ignored",
-          telemetry + ("dpsContactRestrictionId" to it.dpsId),
-        )
-      } ?: run {
-        nomisApiService.getPerson(nomisPersonId = event.personId).also { nomisPerson ->
-          val nomisRestriction = nomisPerson.restrictions.find { it.id == event.visitorRestrictionId } ?: throw IllegalStateException("Restriction ${event.visitorRestrictionId} for person ${event.personId} not found in NOMIS")
+      try {
+        val nomisPerson = nomisApiService.getPerson(nomisPersonId = event.personId)
+        val nomisRestriction = nomisPerson.restrictions.find { it.id == event.visitorRestrictionId } ?: throw IllegalStateException("Restriction ${event.visitorRestrictionId} for person ${event.personId} not found in NOMIS")
+
+        mappingApiService.getByNomisPersonRestrictionIdOrNull(nomisPersonRestrictionId = event.visitorRestrictionId)?.also {
+          telemetry["dpsContactRestrictionId"] = it.dpsId
+          dpsApiService.updateContactRestriction(contactRestrictionId = it.dpsId.toLong(), nomisRestriction.toDpsUpdateContactRestrictionRequest(dpsContactId = event.personId))
+          telemetryClient.trackEvent(
+            "$telemetrySuffix-updated-success",
```

Comment:
Isn't that a prefix?

Reply:
Just this above
` val telemetrySuffix = "contactperson-person-restriction-synchronisation"`

I was trying to reduce a bit of duplication; but maybe I'll go back to the full hardcode string. It is perhaps just confusing